### ISSUE TITLE
[Feature Request] Redirect link of other instances to the chosen instance

### ISSUE THREAD

**Issue by TheFrenchGhosty** [SimonBrazell/privacy-redirect#50]
Invidition support a feature (only for Invidious) named "Always use prefered", it's a feature that redirect every Invidious links of public instances to the chosen instance.

It would be really nice if it was possible to do in Privacy Redirect, either using regex (preferable but might cause some issue in website using the same URL format) or using the list of public instance of each service (how it's implemented in Invidition) either by scraping the list (how it's implemented in Invidition) or by having the list directly in the code.

**Comment by SimonBrazell** [comment]
Just curious what the use case for this feature would be, is it to be used when people share invidious links that you'd prefer redirected to your own or another instance?

Could add some kind of custom redirect rule section to the options menu perhaps, similar to what HTTPS everywhere has? So you could add your own custom redirect rule for any site you like.

**Comment by TheFrenchGhosty** [comment]
>Just curious what the use case for this feature would be, is it to be used when people share invidious links that you'd prefer redirected to your own or another instance?

Mainly this yes, usually when people share links to content on those service they use the official instance. Personally I need this because I pull my YouTube RSS from Invidio.us and watch the videos on my instance.

>Could add some kind of custom redirect rule section to the options menu perhaps, similar to what HTTPS everywhere has? So you could add your own custom redirect rule for any site you like.

Invidition is public domain so you can maybe copy/base your code on its implementation.

**Comment by SimonBrazell** [comment]
No worries, I'll look into it.

**Comment by ToughGuyKunio** [comment]
I was thinking about requesting this feature :smile:  I also use Invidio.us for RSS feeds, and I believe one of the privacy reddits don't allow YouTube-links, so all links are to invidio.us. So there are at least some use cases.

**Comment by 2oq2o5** [comment]
It would be especially interesting in bibliogram, since many instances (including the official one) do not allow RSS subscriptions and all of them are frequently blocked.

**Comment by DJCrashdummy** [comment]
this feature would be awesome especially if also **offline instances** are considered!
because every now and then i'm looking for old links in the history of conversations... and unfortunately some of them are not working because the used instance at that time is not working anymore.

**Comment by TheFrenchGhosty** [comment]
Any news @SimonBrazell ? This is really the only thing missing to Privacy Redirect for me.

**Comment by SimonBrazell** [comment]
Hi @TheFrenchGhosty no progress, I'm busy with life stuff at the moment 🙂

**Comment by TheFrenchGhosty** [comment]
@SimonBrazell Sure, life goes first obviously. I hope everything is fine :)

**Comment by SimonBrazell** [comment]
Yeah all good, just moving house, should be all setup and ready to work on this project again in the next week or two.

**Comment by opusforlife2** [comment]
There's a related problem (maybe it's already mentioned in another issue, but I couldn't find any) that once you're on an instance, such as Invidious or Teddit/Libreddit, any links which direct you elsewhere (such as to Youtube or Old Reddit) aren't intercepted. That means if a user absentmindedly clicks on a link, they might be taken to the main service instead of a front-end.

Do you want me to open a new issue for this, @SimonBrazell?

**Comment by SimonBrazell** [comment]
@opusforlife2 this behaviour is as intended, to allow the "open in YouTube" links to work.

**Comment by opusforlife2** [comment]
But this creates a loophole as I mentioned. With Invidition I just used to temporarily turn off Invidious redirection if I needed to open a YT link. We can do the same here.

**Comment by SimonBrazell** [comment]
This was done on request from other users though, removing it would be seen as a regression by them.

**Comment by opusforlife2** [comment]
Then what alternative do users have in cases like this? I basically have to paste the URL into the address bar each time I want to open such a link, but stay on a privacy preserving website.

**Comment by SimonBrazell** [comment]
Hmm maybe I didn't fully understand your problem, this probably Kore of an issue for reddit right? It probably shouldn't work that way on reddit unless there is an equivalent link on libreddit?

**Comment by SimonBrazell** [comment]
> Do you want me to open a new issue for this

Probably a good idea, sounds like it shouldn't do what you are describing.

**Comment by opusforlife2** [comment]
Libreddit doesn't convert (say) an Old Reddit link in a comment to a Libreddit one, which is a bug/missing feature there, while Teddit does convert such links.

But when it comes to Privacy Redirect, it should intercept any requests to a privacy violating service when redirects are on for that service, shouldn't it?

**Comment by SimonBrazell** [comment]
Yes I was thinking mostly about invidious, which I want to remain the same so the "open in YouTube" links work as even though for the most part you want to avoid YT, sometimes those links are useful if you absolutely must watch the video and its blocked on invidious for some reason. 

Reddit code was possibly done while referring to others so perhaps it has implemented this feature by mistake, please open another issue so this doesn't get lost and I'll fix this up.

Cheers!

**Comment by opusforlife2** [comment]
Should I mention just Reddit or other services as well? (Nitter, Bibliogram, search engines, etc.)

**Comment by SimonBrazell** [comment]
Yeah mention the lot and I'll look into it. Thanks @opusforlife2
